atecc608a_se.h

```diff
@@ -0,0 +1,72 @@
+/**
+ * \file atecc508a_se
+ * \brief Secure element implementation for ATECC508A and ATECC509A
+ */
+
+/*
+ *  Copyright (C) 2019, ARM Limited, All Rights Reserved
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef ATECC608A_SE_H
+#define ATECC608A_SE_H
+
+#include "psa/crypto.h"
+
+#include "atca_status.h"
+#include "atca_devtypes.h"
+#include "atca_iface.h"
+#include "atca_command.h"
+#include "atca_basic.h"
+#include "atca_helpers.h"
+
+#define ASSERT_STATUS(actual, expected)                             \
+    do                                                              \
+    {                                                               \
+        if ((actual) != (expected))                                 \
+        {                                                           \
+            printf("assertion failed at %s:%d "                     \
+                   "(actual=%d expected=%d)\n", __FILE__, __LINE__, \
+                   (int)actual, (int)expected);                     \
```

Comment:
`actual` and `expected` are expressions, not values. This line calculates the expressions again: the way this macro is typically used, this calls the `atcab_xxx` function a second time. Store the values in a local variable and remember to put parentheses around the expression.

```
int ASSERT_STATUS_actual = (actual);
int ASSERT_STATUS_expected = (expected);
if (actual != expected) printf(…, actual, expected);
```